modules/rank-eval/src/test/java/org/elasticsearch/index/rankeval/RankEvalTestHelper.java

```diff
@@ -65,4 +72,31 @@ public static void testHashCodeAndEquals(Object testItem, Object mutation, Objec
         assertThat("testItem copy's hashcode is different from original hashcode", secondCopy.hashCode(),
                 equalTo(testItem.hashCode()));
     }
+
+    /**
+     * Make a deep copy of an object by running it through a BytesStreamOutput
+     * @param original the original object
+     * @param reader a function able to create a new copy of this type
+     * @return a new copy of the original object
+     */
+    public static <T extends Writeable> T copy(T original, Writeable.Reader<T> reader) throws IOException {
+        return copy(original, reader, new NamedWriteableRegistry(Collections.emptyList()));
+    }
+
+    /**
+     * Make a deep copy of an object by running it through a BytesStreamOutput
+     * @param original the original object
+     * @param reader a function able to create a new copy of this type
+     * @param namedWriteableRegistry must be non-empty if the object itself or nested object implement {@link NamedWriteable}
+     * @return a new copy of the original object
+     */
+    public static <T extends Writeable> T copy(T original, Writeable.Reader<T> reader, NamedWriteableRegistry namedWriteableRegistry)
+            throws IOException {
```

Comment:
Capturing another f2f discussion - @cbuescher is already working on pulling this up into a separate helper method.